File: src/ast/SpecialName.java
package ast;

import java.io.StringWriter;

public class SpecialName extends ParentNode {
    private String _specialValue;

    public SpecialName(String specialValue, BaseNode type) {
        super(NodeType.SpecialName, type);
        _specialValue = specialValue;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write(_specialValue);
        child.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/SpecialName.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class SpecialName(private val specialValue: String, type: BaseNode?) : ParentNode(NodeType.SpecialName, type) {
    public override fun printLeft(writer: StringWriter) {
        writer.write(specialValue)
        child!!.print(writer)
    }

}